### PR TITLE
Add widget rewrites + component migration

### PR DESCRIPTION
**Pull request by PoojaB26**
### Description
Provide a brief overview of what this documentation update is about. Explain what sections or topics are being added or revised.

Linear ticket and [magic word](https://linear.app/docs/github#link-prs) Fixes DEVR-330  

## Type of change
- [ ] Typo fix 
- [ ] New feature 
- [ ] Removed outdated references

### CODE REVIEW

File: docs/resources/ui-building-blocks/components/built-in-components/expandable.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+
+# Expandable
```
Comment (pinkeshmars) [inline docs/resources/ui-building-blocks/components/built-in-components/expandable.md:3]: seems like it's WIP

File: docs/resources/ui-building-blocks/components/built-in-components/markdown.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+You could use this widget in various applications like note-taking apps, forums, and blogging platforms. They are particularly popular in technical and coding communities for their ease of formatting code snippets and descriptions.
+
+![img.png](imgs%2Fimg.png)
+
+## Adding Markdown widget
+
+To add a Markdown widget:
+
+1. Open the [Widget Palette](/getting-started/ui-builder/navigation-menu/widget-and-component
```
Comment (pinkeshmars) [inline docs/resources/ui-building-blocks/components/built-in-components/markdown.md:15]: [Widget Palette] link in step 1 and properties panel link in step 2

File: docs/resources/ui-building-blocks/components/built-in-components/sticky-header.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        allowFullScreen
+        allow="clipboard-write">
+    </iframe>
+</div>
+  <figcaption class="centered-caption">Using a search bar as a sticky header widget</figcaption>
+</figure>
+
+Here are the steps to build such an example:
+
+2. First, ensure you have a Column widget on a page. if not, add it. Also, make the Column widget **scrollable** and **Primary**.
```
Comment (pinkeshmars) [inline docs/resources/ui-building-blocks/components/built-in-components/sticky-header.md:92]: Steps number starting from '2'.

**Review by pinkeshmars — CHANGES_REQUESTED** [review]
Hey @PoojaB26 currently I have reviewed widgets under the 'Built-in Components'. Majority are related to fixing links in all pages.

Will add more comments as I review the rest part of this PR :)

**Comment by PoojaB26** [comment]
> Hey @PoojaB26 currently I have reviewed widgets under the 'Built-in Components'. Majority are related to fixing links in all pages.
> 
> Will add more comments as I review the rest part of this PR :)

For hyperlinks, we have to fix in the entire project because there are bunch of dead or empty links in most sections. We can resolve this as a separate task during the beta testing phase. So you can just review the content, and other stuff in this PR or upcoming PRs. @pinkeshmars

**Review by pinkeshmars — CHANGES_REQUESTED** [review]
Hi @PoojaB26 I have added more comments. Could you please take a look?

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+You may want to format large numbers for better readability. Displaying a number like 2,354,000 or 4,356,634,444 instead of 2354000 or 4356634444 enhances the user experience.
+
+For instance, it's clearer to show the population as 1,200,000 rather than 1200000 and currency values like $2K or $5M instead of $2000 or $5000000.
+
+### Types of formatting
+
+Below are the types of formatting that we support:
+
+- **Decimal:** Shows numbers in decimal format (e.g., 1,200,000 and 1.200.000).
```
Comment (pinkeshmars) [inline docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md:154]:
```suggestion
- **Decimal**: Shows numbers in decimal format (e.g., 1,200,000 and 1.200.000).
```

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+- **Percent** : Shows numbers in percentage format (e.g., 28%, 99.99%).
```
Comment (pinkeshmars) [inline docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md:155]:
```suggestion
- **Percent**: Shows numbers in percentage format (e.g., 28%, 99.99%).
```

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+- **Scientific:** Shows numbers in scientific format (e.g., 1e3, 1E6).
+- **Compact** : Shows numbers in compact format (e.g., 2.1K, 2.3M, 5B).
```
Comment (pinkeshmars) [inline docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md:157]:
```suggestion
- **Compact**: Shows numbers in compact format (e.g., 2.1K, 2.3M, 5B).
```

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+For instance, it's clearer to show the population as 1,200,000 rather than 1200000 and currency values like $2K or $5M instead of $2000 or $5000000.
+
+### Types of formatting
+
+Below are the types of formatting that we support:
+
+- **Decimal:** Shows numbers in decimal format (e.g., 1,200,000 and 1.200.000).
+- **Percent** : Shows numbers in percentage format (e.g., 28%, 99.99%).
+- **Scientific:** Shows numbers in scientific format (e.g., 1e3, 1E6).
```
Comment (pinkeshmars) [inline docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md:156]:
```suggestion
- **Scientific**: Shows numbers in scientific format (e.g., 1e3, 1E6).
```

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+- **Compact** : Shows numbers in compact format (e.g., 2.1K, 2.3M, 5B).
+- **Compact Long** : Shows numbers in compact long format (e.g., 2.1 thousand, 2.3 million, 5 billion).
```
Comment (pinkeshmars) [inline docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md:158]:
```suggestion
- **Compact Long**: Shows numbers in compact long format (e.g., 2.1 thousand, 2.3 million, 5 billion).
```

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+- **Custom** : If the given formatting options do not fit your requirement, you can use specify a
```
Comment (pinkeshmars) [inline docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md:159]:
```suggestion
- **Custom**: If the given formatting options do not fit your requirement, you can use specify a 
```

**Review by pinkeshmars — CHANGES_REQUESTED** [review]

**Comment by pinkeshmars** [comment]
@PoojaB26 Just FYI a few of them I directly committed. :)

File: docs/resources/ui-building-blocks/components/built-in-components/expandable.md (hunk 1 above)
Comment (PoojaB26) [reply docs/resources/ui-building-blocks/components/built-in-components/expandable.md:3]: Components are not added to this PR completely, I did whatever I was able to do in the time span. But technically its part of another ticket. https://linear.app/flutterflow/issue/DEVR-403/built-in-components

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/container.md [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+
+In the above example,
+
+- The Transition Point for Color 1 is set at 0, meaning it starts at the very beginning of the
+  gradient.
+- The Transition Point for Color 2 is 0.5, indicating that this color starts transitioning at the
+  halfway point.
+- The Transition Point for Color 3 is 1, which places the start of this color's transition at the
+  end of the gradient.
+  and choosing from Theme Colors, a color picker, or hex codes.
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/resources/ui-building-blocks/widgets/built-in-widgets/container.md:110]: Sorry didnt get you?

[the remark this answers is not in the document]
Comment (pinkeshmars) [reply docs/resources/ui-building-blocks/widgets/built-in-widgets/container.md:110]: Oh, I mean, in the 'Add Color:' point the later part of the description (i.e, after 'by adjusting their transition point...' is placed under the 'The Transition Point for Color 3 is 1...')

[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/resources/ui-building-blocks/widgets/built-in-widgets/container.md:110]: Can you fix this directly please? I'm unable to understand :/

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/container.md [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+  shadow,
+  while a lower value makes the shadow sharper and more defined.
+
+- **Spread:** Controls the **spread radius of the shadow**. **Increasing** this value will **expand** the area
+  that the shadow covers, making it appear larger.
+
+- **Offset X & Offset Y:** These properties set the horizontal (X) and vertical (Y) displacement of
+  the shadow relative to the widget. 
+
+  **Offset X** shifts the shadow horizontally, and **Offset Y**
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/resources/ui-building-blocks/widgets/built-in-widgets/container.md:72]: What do you mean? Like a header? Sorry didnt get you.

File: docs/resources/ui-building-blocks/widgets/widget-commonalities.md [hunk 12]
```diff
@@ -0,0 +1,329 @@
...
+
+<div class="video-container"><iframe src="https://www.loom.
+com/embed/aa1755b1b7b94ef3ac3a72da431d844f?sid=982c1f26-b768-4c8d-ab77-c085219ebab6" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe></div>
+
+
+## Change color
+
+To change color for any widget property:
+
+2. Navigate to a widget property that allows you to set a color.
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/resources/ui-building-blocks/widgets/widget-commonalities.md]: Fixed

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/dividers.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
+
+# Dividers
+
+Add a thin horizontal or vertical line, with padding on either side. Customize the color, width 
+or height, and style of the divider from the Properties Panel. 
+
+Here are the properties in details:
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/resources/ui-building-blocks/widgets/built-in-widgets/dividers.md]: Done

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+either drag it to your desired location on the screen or insert it directly via the widget tree.
+
+![richtext-eg.png](imgs%2Frichtext-eg.png)
+
+When the RichText widget is added to your widget tree, FlutterFlow automatically creates two RichTextSpan child objects. You can modify the text value and styling of each object to create multiple styles within your paragraph.
+
+To modify the RichTextSpan objects, see the quick demo below: 
+
+
+<div style={{
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/resources/ui-building-blocks/widgets/built-in-widgets/text.md:38]: Fixed

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/mouse-region.md [hunk 15]
```diff
@@ -0,0 +1,165 @@
...
+
+The `MouseRegion` widget lets you know whenever the mouse pointer enters or exits from a widget. You could use it to build a user experience (UX), such as animating buttons when a user hovers over them and revealing or hiding menu items when a user hovers over the menu icon.
+
+On this page, you will learn how to [add the MouseRegion widget](#adding-mouseregion-widget), use it to [show/hide elements](#show-hide-elements-using-mouseregion), and [customize](#customizing) it.
+
+## Adding MouseRegion widget
+
+Here are the step-by-step instructions to build such an example:
+
+1. First, click on the **+ Add Widget** and drag the **MouseRegion** widget from the **Base
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/resources/ui-building-blocks/widgets/built-in-widgets/mouse-region.md:12]: Dint get you! What is unusual here?

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/list-grid.md [hunk 16]
```diff
@@ -0,0 +1,286 @@
+---
+title: Lists & Grids
+---
+
+In FlutterFlow, `ListView` and `GridView` are versatile widgets designed for displaying lists and grids 
+of elements, respectively. Both are highly customizable and optimized for dynamic content displays, making them essential for any app that requires scrolling through a collection of items such as images, text, or interactive elements.
+
+## ListView Widget
```
[the remark this answers is not in the document]
Comment (PoojaB26) [reply docs/resources/ui-building-blocks/widgets/built-in-widgets/list-grid.md:8]: yes, seems to be missing, adding to a list of missing items in linear and will add it on a different PR. https://linear.app/flutterflow/issue/DEVR-539/missing-items

**Comment by PoojaB26** [comment]
@pinkeshmars Addressed comments, and left some notes. please check them soon!

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/mouse-region.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (pinkeshmars) [reply docs/resources/ui-building-blocks/widgets/built-in-widgets/mouse-region.md:12]: There was extra spacing after each step. I fixed it directly.

File: docs/resources/ui-building-blocks/widgets/built-in-widgets/container.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (pinkeshmars) [reply docs/resources/ui-building-blocks/widgets/built-in-widgets/container.md:72]: Fixed it directly!

**Comment by pinkeshmars** [comment]
Minor fixes have been pushed. @PoojaB26 this is good to merge now.

**Review by pinkeshmars — APPROVED** [review]

**Pull request merged**